osu.Game/Screens/Menu/MainMenu.cs

```diff
@@ -124,6 +125,7 @@ private void load(BeatmapListingOverlay beatmapListing, SettingsOverlay settings
 
             AddRangeInternal(new[]
             {
+                SeasonalUIConfig.ENABLED ? new MainMenuSeasonalLighting() : Empty(),
```

Comment:
While I was fine with this sort of thing on a temporary branch as I was assuming it to be temporary, I am a little hesitant of putting that sort of stuff upstream. And to be specific, I specifically mean the `ENABLED` checks plastered everywhere.

Thoughts on doing a bit of dependency inversion?:

```csharp
abstract class SeasonalUI
{
    IntroScreen? GetIntroSequence();
    Drawable? CreateLighting();
    MenuSideFlashes? CreateMenuSideFlashes();
    Drawable? CreateKiaiFountains();
    OsuLogo? CreateOsuLogo();
}
```

and then have a `SeasonalUI?` cached at game-level that you can read that stuff from.

I'm basically trying to pre-emptively guard against future "seasonals" there. Unless you think we're never going to do any "seasonals" other than christmas stuff...?

Reply:
sure. i was co considering doing this but felt it'd be best for the next seasonal changes.

Reply:
Fair, probably fine to leave as is until it's necessary then